runtime/gc_realtime/RealtimeMarkingScheme.cpp

```diff
@@ -215,3 +209,89 @@ MM_RealtimeMarkingScheme::incrementalCompleteScan(MM_EnvironmentRealtime *env, u
 	}
 }
 
+uintptr_t
+MM_RealtimeMarkingScheme::scanObject(MM_EnvironmentRealtime *env, uintptr_t item, GC_ObjectScannerState *objectScannerState)
+{
+	uintptr_t scannedPointers = 0;
+	uintptr_t sizeToDo = UDATA_MAX;
+	GC_ObjectScanner *objectScanner = _delegate.getObjectScanner(env, (omrobjectptr_t)item, objectScannerState, SCAN_REASON_PACKET, &sizeToDo);
+	if (NULL != objectScanner) {
+		bool isLeafSlot = false;
+		GC_SlotObject *slotObject;
+		while (NULL != (slotObject = objectScanner->getNextSlot())) {
+			fixupForwardedSlot(slotObject);
+			inlineMarkObjectNoCheck(env, slotObject->readReferenceFromSlot(), isLeafSlot);
+			scannedPointers += 1;
+		}
+	}
```

Comment:
Let's call base scanObject - make it inline on base side if possible. Besides, isLeafSlot is not properly setup.

Reply:
Hm... I guess, you need sizeToDo and scannedPointers. Let me think more.

Reply:
MM_EnvironmentRealtime::_scannedPointerFields  does not see to be used, so you can get rid of it and related getter/setter.

Reply:
Same for _scannedBytes and _scannedObjects.

Hence this whole scanObject becomes identical to base scanObject and you can get rid of this one altogether.

Reply:
Argh, they still differ in return value: base one returns _bytes_ scanned, while realtime one expects scannedPointers to be returned....

Reply:
You can extend env->_markStats and in base scanObject  to maintain _slotsScanned (similar to _objectsScanned). Then in Realtime completeScan loop don't rely on scanObject  return value but on env->_markStats._slotsScanned